frontend/lib/uast/BuilderResult.cpp

```diff
@@ -190,6 +217,18 @@ ID BuilderResult::idToParentId(ID id) const {
   return ID();
 }
 
+// Add getters for additional locations that go from AST or ID to location.
+#define CHPL_LOCATION_MAP(ast__, location__) \
+  Location BuilderResult:: \
+  idTo##location__##Location(ID id, UniqueString path) const { \
+    if (!id) return Location(path); \
+    auto& m = CHPL_ID_LOC_MAP(ast__, location__); \
+    auto it = m.find(id); \
+    return (it != m.end()) ? it->second : Location(path); \
```

Comment:
Does this match the existing behavior of Builder for non-special locations?

Reply:
```cpp
  /** Find the Location for a particular ID.
      Returns a location just to path if none is found. */
  Location idToLocation(ID id, UniqueString path) const;
```